### PR TITLE
clean up implicits for operatorextensions

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala [hunk 1]
```diff
@@ -1,48 +1,74 @@
...
 
 package object operatorextension {
 
-  implicit class ArrayAccessExt(val call: nodes.Call) extends AnyVal {
+  trait Implicits {
+    implicit def toNodeTypeStartersOperatorExtension(cpg: Cpg): NodeTypeStarters = new NodeTypeStarters(cpg)
+    implicit def toArrayAccessExt(call: nodes.Call): ArrayAccessExt = new ArrayAccessExt(call)
+    implicit def toArrayAccessTrav(steps: Steps[nodes.Call]): ArrayAccessTrav = new ArrayAccessTrav(steps)
+    implicit def toAssignmentExt(call: nodes.Call): AssignmentExt = new AssignmentExt(call)
+    implicit def toAssignmentTrav(steps: Steps[nodes.Call]): AssignmentTrav = new AssignmentTrav(steps)
```
Comment (ml86) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala]: This looks to me like we now have the array and assignment steps on all CALL nodes. Is that really the intention?

Comment (fabsx00) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala]: No, that would not be intended. @mpollmeier is this the case?

Comment (fabsx00) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala]: The idea is to have an `isAssignment`, which filters calls for assignments and presents results wrapped in an assignment class, and that that provides methods that are only valid for assignments.

Comment (ml86) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala]: I suggest we create a `nodes.AssignmentCall` which derives from `nodes.Call` and than provide an implicit conversion to `AssignmentExt` for `NodeSteps[nodes.AssignmentCall]`.

Comment (mpollmeier) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala]: ok that makes sense, thanks for catching!

**Pull request merged**